NoSQL/ce-sec/sec/setup-http-proxy-sec.sh

```diff
@@ -10,29 +10,37 @@ fi
 
 mkdir -p /kvroot/proxy/
 
+echo "Creating password"
+openssl rand -out sec/pwd -base64 14
+cp sec/pwd /kvroot/proxy/pwdin
+cp sec/pwd /kvroot/proxy/pwdout
+TMPPWD=$(cat sec/pwd)
+
 echo "Creating USER proxy_user"
 
 java -jar lib/sql.jar -helper-hosts localhost:5000 -store kvstore -security /kvroot/security/user.security <<EOF
-CREATE USER proxy_user IDENTIFIED BY "${KV_PROXY_PWD-ProxyPass@@123}";
+CREATE USER proxy_user IDENTIFIED BY "${TMPPWD}12aB@@";
```

Comment:
Why are you adding `12aB@@` on the end of the password?

Reply:
Because the [password complexity policy](https://docs.oracle.com/en/database/other-databases/nosql-database/22.1/security/password-complexity-policies.html) - I need 2 digits, 2 special characters, ... any ideas?

Reply:
Use `gpg --gen-random --armor 2 14` to generate the password. The `2` is the complexity, and the `14` is how many bytes to output.

Reply:
Here is 2 example of generated passwords
```
gpg --gen-random --armor 2 14
K0ii9I7BErY3ICIOxHs=
JYHF8giJAwoNUs/zBlg=
```
In NoSQL, I have the following error
```
sql-> create user bob identified by "K0ii9I7BErY3ICIOxHs=";
Error handling command create user bob identified by "K0ii9I7BErY3ICIOxHs=": Error: User error in query: CreateUser failed for:
  Password must have at least 2 special characters

sql-> create user bob identified by "JYHF8giJAwoNUs/zBlg=";
Error handling command create user bob identified by "JYHF8giJAwoNUs/zBlg=": Error: User error in query: CreateUser failed for:
  Password must have at least 2 digit numbers

```
This is why I decided to add those characters at the end

Reply:
The command outputs a base64 encoded string, so there will always be an `=` at the end. To be sure you have two special charaters, run the command twice.

Reply:
In my example above, I ran twice, but I have 2 distinct errors
1. at least 2 special characters
2. at least 2 digit numbers

I have executed it multiple times - I was not able to find a pattern

10 executions - 4 is generating an error when creating - I prefer to add something at the end
```
$ gpg --gen-random --armor 2 14
8ng8/2RlzYQGM+jPSI8=  OK
$ gpg --gen-random --armor 2 14
mnWkAaVwwojxshCY8Us= KO
$ gpg --gen-random --armor 2 14
W+4CDE3Wo+E/5u4ICmU= OK
$ gpg --gen-random --armor 2 14
vZ1X7QHLKrbBWqvnFOU= KO
$ gpg --gen-random --armor 2 14
JHAxIXN+kdcjuueLxXM= KO
$ gpg --gen-random --armor 2 14
f+YuKa1qejmamh7ATtw= OK
$ gpg --gen-random --armor 2 14
gvmf1HcDD3mCPzXzDT4= KO
$ gpg --gen-random --armor 2 14
5AaN/C+kmW/o/JC0n+o= OK
$ gpg --gen-random --armor 2 14
GjdGmZu6jtcQA0VV9+4= OK
$ gpg --gen-random --armor 2 14
J+itKbE81Ya3Iu0MeyI= OK
```

Reply:
Sorry, I was being too vague:

```bash
$ TMPPWD="$(gpg --gen-random --armor 2 8)$(gpg --gen-random --armor 2 8)"
$ echo TMPPWD
1QJX/1wxWvmTtQ==B6GGdZI5EXkaUg==
```